larray_editor/arraywidget.py

```diff
@@ -506,9 +506,9 @@ def __init__(self, parent, data_scrollbar):
 
 
 class ArrayEditorWidget(QWidget):
-    def __init__(self, parent, data, readonly=False, bg_value=None, bg_gradient=None, minvalue=None, maxvalue=None):
+    def __init__(self, parent, data=None, readonly=False, bg_value=None, bg_gradient=None, minvalue=None, maxvalue=None):
         QWidget.__init__(self, parent)
-        readonly = np.isscalar(data)
+        readonly = data is not None and np.isscalar(data)
```

Comment:
This seems odd. why do you need readonly=False when data is None?

Reply:
This line is historical. I thought it was there for a good reason so I let it.